### PR TITLE
Fb input constraints

### PR DESCRIPTION
**Pull request by SFulpius**
This pr add some minimum and maximum size for all current inputs we have. See #100 for more details on the limits.

### CODE REVIEW

File: app/src/main/res/values/integers.xml [hunk 1]
```diff
@@ -2,4 +2,10 @@
 <resources>
     <integer name="offer_name_max_size">50</integer>
     <integer name="offer_description_max_size">1000</integer>
+    <integer name="username_min_length">1</integer>
+    <integer name="username_max_length">25</integer>
+    <integer name="user_infos_min_length">0</integer>
+    <integer name="user_infos_max_length">200</integer>
+    <integer name="user_preferences_min_length">0</integer>
+    <integer name="user_preferences_max_length">100</integer>
```
Comment (ElsaWeb) [inline app/src/main/res/values/integers.xml]: So you still needed to add these in Java and XML? If we change our mind about the value, we will have to update both places?

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/ShowOfferActivityTest.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ public void onlyAuthorCanChooseFavorite() {
     }
 
     private void addAnswer(String answer) {
-        onView(withId(R.id.answer_description_))
```
[the remark this answers is not in the document]
Comment (SFulpius) [reply app/src/androidTest/java/ch/epfl/sweng/swenggolf/ShowOfferActivityTest.java]: I have some ideas ...

File: app/src/androidTest/java/ch/epfl/sweng/swenggolf/ProfileActivityTest.java [hunk 3]
```diff
@@ -136,4 +138,15 @@ public void onCancelled(DbError error) {
         DatabaseUser.getUser(vl, user);
 
     }
+
+    @Test
+    public void showErrorMessageWhenUsernameIsTooShort() {
+        onView(withId(R.id.edit_profile)).perform(click());
+        Espresso.closeSoftKeyboard();
+        final ViewInteraction editName = onView(withId(R.id.edit_name));
+        editName.perform(replaceText("")).perform(closeSoftKeyboard());
```
[the remark this answers is not in the document]
Comment (SFulpius) [reply app/src/androidTest/java/ch/epfl/sweng/swenggolf/ProfileActivityTest.java]: I updated it.

File: app/src/main/res/values/integers.xml (hunk 1 above)
Comment (SFulpius) [reply app/src/main/res/values/integers.xml]: These are not used anymore. I deleted these so there will be no confusion which to use in the future.

**Pull request merged**